app/src/ingest_bem_pdfs.py

```diff
@@ -195,44 +195,77 @@ def _split_into_chunks(document: Document, grouped_texts: list[EnrichedText]) ->
     Given EnrichedTexts, convert the text to chunks and add them to the database.
     """
     chunks: list[Chunk] = []
-    for paragraph in grouped_texts:
+    first_paragraph = grouped_texts[0]
+    chunk_text = ""
+    current_heading = first_paragraph.headings
+    page_num = first_paragraph.page_number
+
+    for ind, paragraph in enumerate(grouped_texts):
         assert paragraph.id is not None
         assert paragraph.page_number is not None
-
-        embedding_model = app_config.sentence_transformer
-        token_count = len(embedding_model.tokenizer.tokenize(paragraph.text))
-        if token_count > embedding_model.max_seq_length:
-            # Split the text into chunks of approximately equal length by characters,
-            # which doesn't necessarily mean equal number of tokens, but close enough
-            num_of_splits = math.ceil(token_count / embedding_model.max_seq_length)
-            char_limit_per_split = math.ceil(len(paragraph.text) / num_of_splits)
-            if paragraph.type == TextType.LIST:
-                splits = split_list(paragraph.text, char_limit_per_split)
-            elif paragraph.type == TextType.NARRATIVE_TEXT:
-                splits = split_paragraph(paragraph.text, char_limit_per_split)
-            else:
-                raise ValueError(f"Unexpected element type: {paragraph.type}")
-            logger.info("Split long text into %i chunks: %s", len(splits), splits[0][:120])
+        flat_heading = " ".join([h.title for h in current_heading])
+        current_flat_heading = " ".join([h.title for h in grouped_texts[ind].headings])
+        if flat_heading == current_flat_heading:
+            chunk_text += f"\n{grouped_texts[ind].text}"
```

Comment:
According to [Slack](https://nava.slack.com/archives/C06DP498D1D/p1726088747824089?thread_ts=1726087198.926419&cid=C06DP498D1D), this logic should be in the `group_texts()` function or called from that function. The `_split_into_chunks()` function shouldn't be modified.